libs/sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -2336,6 +2345,19 @@ private static LogicalBinaryExpression.Type getLogicalBinaryOperator(Token token
         }
     }
 
+    private static BitwiseExpression.Type getBitwiseOperator(Token token) {
+        switch (token.getType()) {
+            case SqlBaseLexer.BITWISE_AND:
+                return BitwiseExpression.Type.AND;
+            case SqlBaseLexer.BITWISE_OR:
+                return BitwiseExpression.Type.OR;
+            case SqlBaseLexer.BITWISE_XOR:
+                return BitwiseExpression.Type.XOR;
+            default:
+                throw new IllegalArgumentException("Unsupported operator: " + token.getText());
```

Comment:
Just fyi: Could use switch expressions to avoid the codecov warning and get compiler error if the enum ever gets extended with a new element

Reply:
Could you please elaborate, I'm using switch here but not on an enum, it's a token type which can be any int

Reply:
Oh sorry, I missed that it is on an int, then the switch expression doesn't help. If it were an enum you could have written:

```
        return switch (token.getType()) {
            case SqlBaseLexer.BITWISE_AND -> BitwiseExpression.Type.AND;
            case SqlBaseLexer.BITWISE_OR -> BitwiseExpression.Type.OR;
            case SqlBaseLexer.BITWISE_XOR -> BitwiseExpression.Type.XOR;
        };
```

and it would compile despite missing `default` clause if it handles all cases.